eng/scripts/CodeCheck.ps1

```diff
@@ -0,0 +1,91 @@
+#requires -version 5
+<#
+.SYNOPSIS
+This script runs a quick check for common errors, such as checking that Visual Studio solutions are up to date or that generated code has been committed to source.
+#>
+param(
+    [switch]$ci
+)
+
+$ErrorActionPreference = 'Stop'
+Set-StrictMode -Version 1
+Import-Module -Scope Local -Force "$PSScriptRoot/common.psm1"
+
+$repoRoot = Resolve-Path "$PSScriptRoot/../../"
+
+[string[]] $errors = @()
+
+function LogError([string]$message) {
+    Write-Host -f Red "error: $message"
+    $script:errors += $message
+}
+
+try {
+    #
+    # Solutions
+    #
+
+    if ($ci) {
+        & $PSScriptRoot\..\common\build.ps1 -ci -prepareMachine -build:$false -restore:$false
+    }
+
+    Write-Host "Checking that solutions are up to date"
+
+    Get-ChildItem "$repoRoot/*.sln" -Recurse `
+        | % {
+        Write-Host "  Checking $(Split-Path -Leaf $_)"
+        $slnDir = Split-Path -Parent $_
+        $sln = $_
+        & dotnet sln $_ list `
+            | ? { $_ -ne 'Project(s)' -and $_ -ne '----------' } `
+            | % {
+                $proj = Join-Path $slnDir $_
+                if (-not (Test-Path $proj)) {
+                    LogError "Missing project. Solution references a project which does not exist: $proj. [$sln] "
+                }
+            }
```

Comment:
Suggest being more consistent with indentation in the two `foreach` blocks. (This is a bit harder to read that in needs to be.) The first `Foreach-Object` block has no indentation but the second has. It's fine if either the close curlies aren't laid out perfectly (as they aren't for the outer line continuation).

FYI I'm going to put `dotnet sln $_ list ...` into a function in my PowerShell profile 😃

Reply:
I too which the indentation was better. The auto-formatter in VSCode does this, so I'll manually undo it for now.